platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -944,6 +955,17 @@ def _parse_vpd_data(self, filename):
 
         return result
 
+    def _read_model_from_vpd(self):
+        """
+        Returns if model number should be returned from VPD file
+
+        Returns:
+            Returns True if spectrum version is higher than Spectrum-4 according to sku number
+        """
+        sku = device_info.get_hwsku()
+        sku_num = re.search('[0-9]{4}', sku).group()
```

Comment:
I would suggest adding error-handling logic in case sku_num isn't available.

Reply:
Good catch, thanks. Handled.